### PR TITLE
ths: new cask

### PR DESCRIPTION
**Pull request by TsanBoi**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/ths.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+cask "ths" do
```
Comment (razvanazamfirei) [inline Casks/ths.rb:1]: Could you verify that this follows our guidelines for cask [tokens](https://docs.brew.sh/Cask-Cookbook#converting-to-ascii)?

File: Casks/ths.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  version "4.0.3"
+  sha256 "757c65787375723eda5b35005cc0d37dd848c49d3e5d79b5a8bc963fa2257617"
+
+  url "https://resource.thsi.cn/soft/ThsMac#{version}.dmg",
+      verified: "resource.thsi.cn/soft/"
+  name "同花顺"
+  name "Straight Flush"
```
Comment (razvanazamfirei) [inline Casks/ths.rb:8]: Is this an accurate translation?

**Review by razvanazamfirei — COMMENTED** [review]
Hi @Vulxan, thank you for submitting this. I had a couple of questions regarding naming, but otherwise it looks great!

File: Casks/ths.rb (hunk 1 above)
Comment (Vulxan) [reply Casks/ths.rb:1]: "ths" corresponds to the three initials of the Chinese pinyin——tong hua shun😂

File: Casks/ths.rb (hunk 2 above)
Comment (Vulxan) [reply Casks/ths.rb:8]: Yes, the name seems more suitable.

File: Casks/ths.rb (hunk 1 above)
Comment (razvanazamfirei) [reply Casks/ths.rb:1]: Thanks! :) Just double-checking with the experts.

**Review by razvanazamfirei — APPROVED** [review]

**Pull request merged**